src/invidious/routes/channels.cr

```diff
@@ -0,0 +1,149 @@
+      items.reject! &.paid
+    end
+
+    templated "channel"
+  end
+
+  def playlists(env)
+    data = self.fetch_basic_information(env)
+    if !data.is_a?(Tuple)
+      return data
+    end
+    locale, user, subscriptions, continuation, ucid, channel = data
+
+    sort_options = {"last", "oldest", "newest"}
+    sort_by = env.params.query["sort_by"]?.try &.downcase
+    sort_by ||= "last"
+
+    if channel.auto_generated
+      return env.redirect "/channel/#{channel.ucid}"
+    end
+
+    items, continuation = fetch_channel_playlists(channel.ucid, channel.author, continuation, sort_by)
+    items = items.select { |item| item.is_a?(SearchPlaylist) }.map { |item| item.as(SearchPlaylist) }
+    items.each { |item| item.author = "" }
+
+    templated "playlists"
+  end
+
+  def community(env)
+    data = self.fetch_basic_information(env)
+    if !data.is_a?(Tuple)
+      return data
+    end
+    locale, user, subscriptions, continuation, ucid, channel = data
+
+    thin_mode = env.params.query["thin_mode"]? || env.get("preferences").as(Preferences).thin_mode
+    thin_mode = thin_mode == "true"
+
+    continuation = env.params.query["continuation"]?
+    # sort_by = env.params.query["sort_by"]?.try &.downcase
+
+    if !channel.tabs.includes? "community"
+      return env.redirect "/channel/#{channel.ucid}"
+    end
+
+    begin
+      items = JSON.parse(fetch_channel_community(ucid, continuation, locale, "json", thin_mode))
+    rescue ex : InfoException
+      env.response.status_code = 500
+      error_message = ex.message
+    rescue ex
+      return error_template(500, ex)
+    end
+
+    templated "community"
+  end
+
+  def about(env)
+    data = self.fetch_basic_information(env)
+    if !data.is_a?(Tuple)
+      return data
+    end
+    locale, user, subscriptions, continuation, ucid, channel = data
+
+    env.redirect "/channel/#{ucid}"
+  end
+
+  # Redirects brand url channels to a normal /channel/:ucid route
+  def brand_redirect(env)
+    locale = LOCALES[env.get("preferences").as(Preferences).locale]?
+
+    begin
+      resolved_url = request_youtube_api_resolve_url("https://youtube.com#{env.request.path}#{env.params.query.size > 0 ? "?#{env.params.query}" : ""}")
+    rescue ex : InfoException
+      raise InfoException.new("This channel does not exist.")
+    end
+
+    ucid = resolved_url["endpoint"]["browseEndpoint"]["browseId"]
+
+    selected_tab = env.request.path.split("/")[-1]
+    if ["home", "videos", "playlists", "community", "channels", "about"].includes? selected_tab
+      url = "/channel/#{ucid}/#{selected_tab}"
+    else
+      url = "/channel/#{ucid}"
+    end
+
+    env.redirect url
+  end
+
+  private def fetch_basic_information(env)
+    locale = LOCALES[env.get("preferences").as(Preferences).locale]?
+
+    user = env.get? "user"
+    if user
+      user = user.as(User)
+      subscriptions = user.subscriptions
+    end
+    subscriptions ||= [] of String
+
+    ucid = env.params.url["ucid"]
```

Comment:
You should add a verificatin after this that we have an UCID provided. if we didn't, we should resolve in here too (e.g: acessing `/channel/kevin/videos` won't go through `brand_redirect`, and we still won't have a valid UCID to work with).

Reply:
That shouldn't be an issue :) An invalid channel URL always [returns an error json response](https://bpa.st/NOKQ) which automatically raises an `InfoException` as seen here:

https://github.com/iv-org/invidious/blob/a88e5b3b84ae40d2c39f42d80afa79f1427e096f/src/invidious/helpers/youtube_api.cr#L126-L133

This is then catched and handled in the `brand_redirect` method.

Reply:
The problem is that `/channel/<name>/videos` is a valid URL, but we won't have the UCID for it.

Reply:
Ah oops looks like I completely misread your original message, sorry about that. I still fail to see how this is an problem though, `/channel/:name` isn't a valid endpoint even on YouTube so there's absolutely no reason to pass it into `brand_redirect`. Furthermore, the get_about_info function already verifies ucids and returns an InfoException when the channel doesn't exist:

Reply:
> I still fail to see how this is a problem though, `/channel/:name` isn't a valid endpoint even on YouTube so there's absolutely no reason to pass it into `brand_redirect`.

Oooh, right, for some reason I thought that it was a valid endpoint...